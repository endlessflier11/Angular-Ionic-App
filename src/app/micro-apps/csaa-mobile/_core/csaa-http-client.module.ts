import { NgModule } from '@angular/core';
import { AngularHttpAdapter } from './services/http/adapters/angular-http.adapter';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { IonicHttpAdapter } from './services/http/adapters/ionic-http.adapter';
import { HTTP } from '@ionic-native/http/ngx';
import { UNI_HTTP_ADAPTER } from './services/http/uni-http.model';

@NgModule({
  imports: [HttpClientModule],
  providers: [
    HTTP,
    { provide: UNI_HTTP_ADAPTER, multi: true, useClass: AngularHttpAdapter, deps: [HttpClient] },
    {
      provide: UNI_HTTP_ADAPTER,
      multi: true,
      useClass: IonicHttpAdapter,
      deps: [HTTP],
    },
  ],
  exports: [HttpClientModule],
})
export class CsaaHttpClientModule {}
