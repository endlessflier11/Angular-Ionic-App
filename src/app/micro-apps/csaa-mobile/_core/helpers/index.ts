export * from './router.helpers';
export * from './misc.helpers';
export * from './async.helper';
export * from './error.helper';
export * from './string.helper';
