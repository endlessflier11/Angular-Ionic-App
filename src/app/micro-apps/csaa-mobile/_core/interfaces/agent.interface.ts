export interface Agent {
  name: string;
  phone: string;
  email: string;
  isService: boolean;
}
