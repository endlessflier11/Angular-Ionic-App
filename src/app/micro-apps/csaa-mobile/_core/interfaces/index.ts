export * from './auth.interface';
export * from './router';
export * from './http.interface';
export * from './vehicle.interface';
export * from './ionic-lifecycle.interface';
export * from './analytics.interface';
export * from './policy.interface';
export * from './payment.interface';
export * from './config.interface';
export * from './ngxs-store.interface';
