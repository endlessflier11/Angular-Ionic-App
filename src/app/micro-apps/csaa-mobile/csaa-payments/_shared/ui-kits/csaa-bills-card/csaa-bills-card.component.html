<csaa-skeleton-card *ngIf="loading"></csaa-skeleton-card>
<ion-card *ngIf="!loading && relevantBills.length" class="csaa-card">
  <ion-card-header>
    <ion-card-title>Bills</ion-card-title>
  </ion-card-header>
  <ion-card-content class="csaa-item-list">
    <ion-list>
      <ion-item
        detail
        *ngFor="let bill of relevantBills"
        (click)="handlePolicyClick(bill)"
        class="ion-text-wrap"
      >
        <ion-label>
          <h3>{{ bill?.policyNumber }}</h3>
          <span class="subtitle">
            <csaa-policy-subtitles [policySubtitle]="bill?.subtitle"></csaa-policy-subtitles>
          </span>
          <label class="bill-data">
            Bill {{ bill?.amount | currency: 'USD':'symbol':'1.2-2' }}<br />
            <ng-container *ngIf="autoPayEnrollmentStatusMap[bill?.policyNumber]; else billDate">
              AutoPay <ion-text color="success" class="bold">ON</ion-text>
              <br />
              Sent
              {{ getDocumentProcessDate(bill) | date: 'mediumDate':timezone }}
            </ng-container>
            <ng-template #billDate>
              Sent
              {{
                bill?.transactionDate?.split('-').slice(0, 3).join('-')
                  | date: 'mediumDate':timezone
              }}
            </ng-template>
          </label>
        </ion-label>

        <csaa-theme-icon slot="end" [type]="bill?.policyType" [size]="40"> </csaa-theme-icon>
      </ion-item>
    </ion-list>
  </ion-card-content>
</ion-card>
