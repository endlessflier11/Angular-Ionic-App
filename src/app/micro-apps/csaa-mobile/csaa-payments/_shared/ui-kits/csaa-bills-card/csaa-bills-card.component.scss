@import '../../../../_scss/variables';
.bill-data {
  color: $csaa-lighter-color;
  display: block;
  margin-top: 5px;
}

.bold {
  font-weight: bold;
}
