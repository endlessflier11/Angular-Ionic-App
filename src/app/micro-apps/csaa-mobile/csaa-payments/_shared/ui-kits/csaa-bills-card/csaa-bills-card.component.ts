import { Component, EventEmitter, Input, Output } from '@angular/core';
import { add, isBefore, parseISO } from 'date-fns';
import { AutoPayEnrollmentResponse, Bill, PolicyType } from '../../../../_core/interfaces';
import { PolicyDocument } from '../../../../_core/interfaces/document.interface';

enum DocumentName {
  INSTALLMENT_BILL = 'Installment Bill',
  AUTOPAY_SCHEDULE = 'Auto Pay Schedule',
}

@Component({
  selector: 'csaa-csaa-bills-card',
  templateUrl: './csaa-bills-card.component.html',
  styleUrls: ['./csaa-bills-card.component.scss'],
})
export class CsaaBillsCardComponent {
  @Input() bills: Bill[] = [];
  @Input() autoPayEnrollmentStatusMap: {
    [policyNumber: string]: AutoPayEnrollmentResponse | null;
  } = {};
  @Input() isPaidInFullStatusMap: { [policyNumber: string]: boolean } = {};
  @Input() availableDocumentsMap: { [policyNumber: string]: PolicyDocument[] } = {};
  @Input() loading = false;
  @Input() timezone: string = undefined;
  @Output() billClick = new EventEmitter<Bill>();
  // eslint-disable-next-line @typescript-eslint/naming-convention
  PolicyType = PolicyType;

  public get relevantBills(): Bill[] {
    // return an array of bills whose process dates are within one calender month
    // and remaining balance is zero
    return this.bills.filter((b) => {
      if (this.hasAutopayEnrolled(b)) {
        return this.hasAutoPayScheduleDocument(b);
      }

      // Wait! Document shown when display condition passes exists?
      // MM-3081: criteria 2 - Installment Bill document exists
      // and (sort by processDate src/app/micro-apps/csaa-mobile/_core/services/policy.service.ts:335)
      const doc = this.getInstallmentBillDocument(b);
      if (
        !doc ||
        // eslint-disable-next-line max-len
        !CsaaBillsCardComponent.isBeforeOneCalendarMonth(parseISO(doc.processDate)) // MM-3081: criteria 3 - safeguard against displaying an outdated bill.
      ) {
        return false;
      }

      return (
        !this.isPaidInFull(b) ||
        CsaaBillsCardComponent.isBeforeOneCalendarMonth(parseISO(b.dueDate)) // MM-3081: criteria 1 - dueDate is withing one calendar month
      );
    });
  }

  private static isBeforeOneCalendarMonth(txnDate: Date): boolean {
    const nextMonth = add(txnDate, { months: 1 });
    return isBefore(new Date(), nextMonth);
  }

  handlePolicyClick(bill: Bill) {
    // TODO: Segment event TBD
    this.billClick.emit(bill);
  }

  public getDocumentProcessDate(bill: Bill): string {
    return this.availableDocumentsMap[bill.policyNumber].find(
      (n) => n.docName === 'Auto Pay Schedule'
    ).processDate;
  }

  private isPaidInFull(bill: Bill): boolean {
    return Boolean(this.isPaidInFullStatusMap[bill.policyNumber]);
  }

  private hasAutopayEnrolled(bill: Bill): boolean {
    return Boolean(this.autoPayEnrollmentStatusMap[bill.policyNumber]?.autoPay);
  }

  private hasAutoPayScheduleDocument(bill: Bill): boolean {
    return this.availableDocumentsMap[bill.policyNumber]?.some(
      (d) => d.docName === DocumentName.AUTOPAY_SCHEDULE
    );
  }

  private getInstallmentBillDocument(bill: Bill): PolicyDocument | null {
    return this.availableDocumentsMap[bill.policyNumber]?.find(
      (d) => d.docName === DocumentName.INSTALLMENT_BILL
    );
  }
}
