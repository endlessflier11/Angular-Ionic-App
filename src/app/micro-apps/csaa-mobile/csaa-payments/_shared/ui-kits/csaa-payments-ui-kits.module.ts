import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaPaymentHistoryCardComponent } from './csaa-payment-history-card/csaa-payment-history-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentHistorySelectCardComponent } from './csaa-payment-history-select-card/csaa-payment-history-select-card.component';
import { CsaaPaymentSucceededCardComponent } from './csaa-payment-succeeded-card/csaa-payment-succeeded-card.component';
import { CsaaMakePaymentCardComponent } from './csaa-make-payment-card/csaa-make-payment-card.component';
import { CsaaPayAllPoliciesCardComponent } from './csaa-pay-all-policies-card/csaa-pay-all-policies-card.component';
import { PaymentTermsConditionsComponent } from './payment-terms-conditions/payment-terms-conditions.component';
import { AutopayTermsConditionsComponent } from './autopay-terms-conditions/autopay-terms-conditions.component';
import { CsaaBillsCardComponent } from './csaa-bills-card/csaa-bills-card.component';
import { AutopayDiscountCardComponent } from './autopay-discount-card/autopay-discount-card.component';

const SHARED_COMPONENTS = [
  CsaaPaymentHistoryCardComponent,
  CsaaPaymentHistorySelectCardComponent,
  CsaaMakePaymentCardComponent,
  CsaaPaymentSucceededCardComponent,
  CsaaPayAllPoliciesCardComponent,
  PaymentTermsConditionsComponent,
  AutopayTermsConditionsComponent,
  CsaaBillsCardComponent,
  AutopayDiscountCardComponent,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule, IonicModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaPaymentsUiKitsModule {}
