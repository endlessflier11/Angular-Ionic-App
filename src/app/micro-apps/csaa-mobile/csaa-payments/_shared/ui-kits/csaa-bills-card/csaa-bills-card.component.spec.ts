import { ComponentFixture, TestBed } from '@angular/core/testing';
import { IonicModule } from '@ionic/angular';

import { CsaaBillsCardComponent } from './csaa-bills-card.component';
import { CommonModule } from '@angular/common';
import { UiKitsModule } from '../../../../_core/ui-kits/ui-kits.module';
import { By, PageTestingModule, StoreTestBuilder } from '@app/testing';
import { CsaaPaymentsUiKitsModule } from '../csaa-payments-ui-kits.module';
import { Store } from '@ngxs/store';
import { add, format, parse } from 'date-fns';
import { AutoPayEnrollmentResponse } from '../../../../_core/interfaces';
import { PolicyDocument } from '../../../../_core/interfaces/document.interface';

describe('CsaaBillsCardComponent', () => {
  const transactionDate = parse('2021-01-27', 'yyyy-MM-dd', new Date());
  const policyNumber = 'CAAC910026006';

  let component: CsaaBillsCardComponent;
  let fixture: ComponentFixture<CsaaBillsCardComponent>;
  let store: Store;

  const setupComponent = (beforeComponentCreate?: () => void) => {
    TestBed.configureTestingModule({
      imports: [
        CommonModule,
        IonicModule,
        UiKitsModule,
        CsaaPaymentsUiKitsModule,
        PageTestingModule.withConfig({
          providesAnalytics: true,
          providesStorage: true,
          providesConfig: true,
        }),
      ],
    }).compileComponents();

    store = TestBed.inject(Store);
    StoreTestBuilder.withDefaultMocks().resetStateOf(store);

    beforeComponentCreate && beforeComponentCreate.call(beforeComponentCreate);

    fixture = TestBed.createComponent(CsaaBillsCardComponent);
    component = fixture.componentInstance;
    component.bills = [
      {
        amount: '23.50',
        billedTo: 'JEAN BRANDOS',
        dueDate: format(transactionDate, 'yyyy-MM-dd'), // '2021-03-09',
        paymentPlan: 'standardHO',
        policyEffectiveDate: '2020-09-09',
        policyExpirationDate: '2021-09-09',
        policyNumber,
        policyType: 1,
        priorBalance: '0.00',
        referenceNumber: '130081',
        subtitle: '610 N 1st St',
        totalAmountDue: '119.50',
        transactionDate: '2021-01-27',
      },
    ];
    component.autoPayEnrollmentStatusMap = {};
    component.isPaidInFullStatusMap = {};
    component.availableDocumentsMap = {};
    fixture.detectChanges();
  };

  afterEach(() => {
    jest.useRealTimers();
  });

  it('should create', () => {
    setupComponent();
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    setupComponent();
    jest.useFakeTimers('modern');
    jest.setSystemTime(new Date(transactionDate));
    component.availableDocumentsMap = {
      [policyNumber]: [
        {
          processDate: format(transactionDate, 'yyyy-MM-dd'),
          docName: 'Installment Bill',
        } as PolicyDocument,
      ],
    };
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  describe('AutoPay Not Enrolled', () => {
    it('should match snapshot >> before one calendar month of transaction not paid in full', () => {
      jest.useFakeTimers('modern');
      jest.setSystemTime(new Date(add(transactionDate, { days: 25 }).getTime()));

      setupComponent(() => {
        StoreTestBuilder.withDefaultMocks()
          .patchCustomerPaymentDataAtIndex(0, { remainingPremium: 5000 })
          .resetStateOf(store);
      });

      component.isPaidInFullStatusMap = {
        [policyNumber]: false,
      };
      component.availableDocumentsMap = {
        [policyNumber]: [
          {
            processDate: format(transactionDate, 'yyyy-MM-dd'),
            docName: 'Installment Bill',
          } as PolicyDocument,
        ],
      };
      fixture.detectChanges();

      expect(fixture.debugElement.query(By.cssAndText('ion-card', policyNumber))).toBeTruthy();
      expect(fixture).toMatchSnapshot();
    });

    it('should match snapshot >> before one calendar month of transaction and paid in full', () => {
      jest.useFakeTimers('modern');
      jest.setSystemTime(new Date(add(transactionDate, { days: 25 }).getTime()));

      setupComponent(() => {
        StoreTestBuilder.withDefaultMocks()
          .patchCustomerPaymentDataAtIndex(0, { remainingPremium: 0 })
          .resetStateOf(store);
      });

      component.isPaidInFullStatusMap = {
        [policyNumber]: true,
      };
      component.availableDocumentsMap = {
        [policyNumber]: [
          {
            processDate: format(transactionDate, 'yyyy-MM-dd'),
            docName: 'Installment Bill',
          } as PolicyDocument,
        ],
      };
      fixture.detectChanges();

      expect(fixture.debugElement.query(By.cssAndText('ion-card', policyNumber))).toBeTruthy();
      expect(fixture).toMatchSnapshot();
    });

    it('should match snapshot >> after one calendar month of transaction and paid in full', () => {
      jest.useFakeTimers('modern');
      jest.setSystemTime(new Date(add(transactionDate, { months: 1 }).getTime()));

      setupComponent(() => {
        StoreTestBuilder.withDefaultMocks()
          .patchCustomerPaymentDataAtIndex(0, { remainingPremium: 0 })
          .resetStateOf(store);
      });

      component.isPaidInFullStatusMap = {
        [policyNumber]: true,
      };
      component.availableDocumentsMap = {
        [policyNumber]: [
          {
            processDate: format(add(transactionDate, { months: 1 }), 'yyyy-MM-dd'),
            docName: 'Installment Bill',
          } as PolicyDocument,
        ],
      };
      fixture.detectChanges();

      expect(fixture.debugElement.query(By.cssAndText('ion-card', policyNumber))).toBeFalsy();
      expect(fixture).toMatchSnapshot();
    });

    it('should match snapshot >> after one calendar month of transaction and not paid in full', () => {
      jest.useFakeTimers('modern');
      jest.setSystemTime(new Date(add(transactionDate, { months: 1 })));

      setupComponent(() => {
        StoreTestBuilder.withDefaultMocks()
          .patchCustomerPaymentDataAtIndex(0, { remainingPremium: 5000 })
          .resetStateOf(store);
      });

      component.isPaidInFullStatusMap = {
        [policyNumber]: false,
      };
      component.availableDocumentsMap = {
        [policyNumber]: [
          {
            processDate: format(new Date(), 'yyyy-MM-dd'),
            docName: 'Installment Bill',
          } as PolicyDocument,
          // {processDate: format(transactionDate, 'yyyy-MM-dd'), docName: 'Installment Bill'} as PolicyDocument
        ],
      };
      fixture.detectChanges();

      expect(fixture.debugElement.query(By.cssAndText('ion-card', policyNumber))).toBeTruthy();
      expect(fixture).toMatchSnapshot();
    });
  });

  describe('AutoPay Enrolled', () => {
    it('should match snapshot >> after one calendar month of transaction and paid in full', () => {
      jest.useFakeTimers('modern');
      jest.setSystemTime(new Date(add(transactionDate, { months: 1 }).getTime()));

      setupComponent(() => {
        StoreTestBuilder.withDefaultMocks()
          .patchCustomerPaymentDataAtIndex(0, { remainingPremium: 5000 })
          .resetStateOf(store); // recipe for hide --- (1)
      });

      component.autoPayEnrollmentStatusMap = {
        [policyNumber]: { autoPay: true } as unknown as AutoPayEnrollmentResponse,
      };
      component.isPaidInFullStatusMap = {
        [policyNumber]: true, // recipe for hide --- (2)
      };
      component.availableDocumentsMap = {
        [policyNumber]: [
          {
            externalURI: 'policy/documentsecure/AAAA...',
            docName: 'Auto Pay Schedule',
            docType: 'payment',
            description: 'Auto Pay Schedule',
            processDate: '2020-10-09',
            agreementEffectiveDate: '2020-10-09',
            oid: 'b0a84da26b8d1b8d42e74...',
          } as PolicyDocument,
        ],
      };
      fixture.detectChanges();

      expect(fixture.debugElement.query(By.cssAndText('ion-card', policyNumber))).toBeTruthy(); // but shown --- (3)
      expect(fixture).toMatchSnapshot();
    });
    it('should match screenshot if payment plan is mortgageeHO', () => {
      component.bills = [
        {
          amount: '23.50',
          billedTo: 'JEAN BRANDOS',
          dueDate: '2021-03-09',
          paymentPlan: 'mortgageeHO',
          policyEffectiveDate: '2020-09-09',
          policyExpirationDate: '2021-09-09',
          policyNumber: 'MTH4910017142',
          policyType: 1,
          priorBalance: '0.00',
          referenceNumber: '130081',
          subtitle: '610 N 1st St',
          totalAmountDue: '119.50',
          transactionDate: '2021-01-27',
        },
      ];
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });
  });
});
