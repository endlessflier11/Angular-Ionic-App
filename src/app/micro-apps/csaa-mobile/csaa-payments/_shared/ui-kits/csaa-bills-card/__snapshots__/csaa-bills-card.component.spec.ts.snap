// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsaaBillsCardComponent AutoPay Enrolled should match screenshot if payment plan is mortgageeHO 1`] = `
<csaa-csaa-bills-card
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  autoPayEnrollmentStatusMap={[Function Object]}
  availableDocumentsMap={[Function Object]}
  billClick={[Function EventEmitter_]}
  bills={[Function Array]}
  isPaidInFullStatusMap={[Function Object]}
  loading="false"
  timezone="undefined"
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        Bills
      </ion-card-title>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item
          class="ion-text-wrap"
          detail=""
        >
          <ion-label>
            <h3>
              CAAC910026006
            </h3>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  610 N 1st St
                </span>
              </csaa-policy-subtitles>
            </span>
            <label
              class="bill-data"
            >
               Bill $23.50
              <br />
               AutoPay 
              <ion-text
                class="bold"
                color="success"
              >
                ON
              </ion-text>
              <br />
               Sent Oct 9, 2020 
            </label>
          </ion-label>
          <csaa-theme-icon
            slot="end"
          >
            <img
              src="/assets/csaa-mobile/vectors/icn-Home.svg"
              style="width: 40px; height: 40px;"
            />
          </csaa-theme-icon>
        </ion-item>
      </ion-list>
    </ion-card-content>
  </ion-card>
</csaa-csaa-bills-card>
`;

exports[`CsaaBillsCardComponent AutoPay Enrolled should match snapshot >> after one calendar month of transaction and paid in full 1`] = `
<csaa-csaa-bills-card
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  autoPayEnrollmentStatusMap={[Function Object]}
  availableDocumentsMap={[Function Object]}
  billClick={[Function EventEmitter_]}
  bills={[Function Array]}
  isPaidInFullStatusMap={[Function Object]}
  loading="false"
  timezone="undefined"
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        Bills
      </ion-card-title>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item
          class="ion-text-wrap"
          detail=""
        >
          <ion-label>
            <h3>
              CAAC910026006
            </h3>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  610 N 1st St
                </span>
              </csaa-policy-subtitles>
            </span>
            <label
              class="bill-data"
            >
               Bill $23.50
              <br />
               AutoPay 
              <ion-text
                class="bold"
                color="success"
              >
                ON
              </ion-text>
              <br />
               Sent Oct 9, 2020 
            </label>
          </ion-label>
          <csaa-theme-icon
            slot="end"
          >
            <img
              src="/assets/csaa-mobile/vectors/icn-Home.svg"
              style="width: 40px; height: 40px;"
            />
          </csaa-theme-icon>
        </ion-item>
      </ion-list>
    </ion-card-content>
  </ion-card>
</csaa-csaa-bills-card>
`;

exports[`CsaaBillsCardComponent AutoPay Not Enrolled should match snapshot >> after one calendar month of transaction and not paid in full 1`] = `
<csaa-csaa-bills-card
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  autoPayEnrollmentStatusMap={[Function Object]}
  availableDocumentsMap={[Function Object]}
  billClick={[Function EventEmitter_]}
  bills={[Function Array]}
  isPaidInFullStatusMap={[Function Object]}
  loading="false"
  timezone="undefined"
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        Bills
      </ion-card-title>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item
          class="ion-text-wrap"
          detail=""
        >
          <ion-label>
            <h3>
              CAAC910026006
            </h3>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  610 N 1st St
                </span>
              </csaa-policy-subtitles>
            </span>
            <label
              class="bill-data"
            >
               Bill $23.50
              <br />
               Sent Jan 27, 2021 
            </label>
          </ion-label>
          <csaa-theme-icon
            slot="end"
          >
            <img
              src="/assets/csaa-mobile/vectors/icn-Home.svg"
              style="width: 40px; height: 40px;"
            />
          </csaa-theme-icon>
        </ion-item>
      </ion-list>
    </ion-card-content>
  </ion-card>
</csaa-csaa-bills-card>
`;

exports[`CsaaBillsCardComponent AutoPay Not Enrolled should match snapshot >> after one calendar month of transaction and paid in full 1`] = `
<csaa-csaa-bills-card
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  autoPayEnrollmentStatusMap={[Function Object]}
  availableDocumentsMap={[Function Object]}
  billClick={[Function EventEmitter_]}
  bills={[Function Array]}
  isPaidInFullStatusMap={[Function Object]}
  loading="false"
  timezone="undefined"
>
</csaa-csaa-bills-card>
`;

exports[`CsaaBillsCardComponent AutoPay Not Enrolled should match snapshot >> before one calendar month of transaction and paid in full 1`] = `
<csaa-csaa-bills-card
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  autoPayEnrollmentStatusMap={[Function Object]}
  availableDocumentsMap={[Function Object]}
  billClick={[Function EventEmitter_]}
  bills={[Function Array]}
  isPaidInFullStatusMap={[Function Object]}
  loading="false"
  timezone="undefined"
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        Bills
      </ion-card-title>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item
          class="ion-text-wrap"
          detail=""
        >
          <ion-label>
            <h3>
              CAAC910026006
            </h3>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  610 N 1st St
                </span>
              </csaa-policy-subtitles>
            </span>
            <label
              class="bill-data"
            >
               Bill $23.50
              <br />
               Sent Jan 27, 2021 
            </label>
          </ion-label>
          <csaa-theme-icon
            slot="end"
          >
            <img
              src="/assets/csaa-mobile/vectors/icn-Home.svg"
              style="width: 40px; height: 40px;"
            />
          </csaa-theme-icon>
        </ion-item>
      </ion-list>
    </ion-card-content>
  </ion-card>
</csaa-csaa-bills-card>
`;

exports[`CsaaBillsCardComponent AutoPay Not Enrolled should match snapshot >> before one calendar month of transaction not paid in full 1`] = `
<csaa-csaa-bills-card
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  autoPayEnrollmentStatusMap={[Function Object]}
  availableDocumentsMap={[Function Object]}
  billClick={[Function EventEmitter_]}
  bills={[Function Array]}
  isPaidInFullStatusMap={[Function Object]}
  loading="false"
  timezone="undefined"
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        Bills
      </ion-card-title>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item
          class="ion-text-wrap"
          detail=""
        >
          <ion-label>
            <h3>
              CAAC910026006
            </h3>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  610 N 1st St
                </span>
              </csaa-policy-subtitles>
            </span>
            <label
              class="bill-data"
            >
               Bill $23.50
              <br />
               Sent Jan 27, 2021 
            </label>
          </ion-label>
          <csaa-theme-icon
            slot="end"
          >
            <img
              src="/assets/csaa-mobile/vectors/icn-Home.svg"
              style="width: 40px; height: 40px;"
            />
          </csaa-theme-icon>
        </ion-item>
      </ion-list>
    </ion-card-content>
  </ion-card>
</csaa-csaa-bills-card>
`;

exports[`CsaaBillsCardComponent should match snapshot 1`] = `
<csaa-csaa-bills-card
  PolicyType={[Function Object]}
  __ngContext__={[Function LRootView]}
  autoPayEnrollmentStatusMap={[Function Object]}
  availableDocumentsMap={[Function Object]}
  billClick={[Function EventEmitter_]}
  bills={[Function Array]}
  isPaidInFullStatusMap={[Function Object]}
  loading="false"
  timezone="undefined"
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        Bills
      </ion-card-title>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item
          class="ion-text-wrap"
          detail=""
        >
          <ion-label>
            <h3>
              CAAC910026006
            </h3>
            <span
              class="subtitle"
            >
              <csaa-policy-subtitles>
                <span
                  class="subtitle"
                >
                  610 N 1st St
                </span>
              </csaa-policy-subtitles>
            </span>
            <label
              class="bill-data"
            >
               Bill $23.50
              <br />
               Sent Jan 27, 2021 
            </label>
          </ion-label>
          <csaa-theme-icon
            slot="end"
          >
            <img
              src="/assets/csaa-mobile/vectors/icn-Home.svg"
              style="width: 40px; height: 40px;"
            />
          </csaa-theme-icon>
        </ion-item>
      </ion-list>
    </ion-card-content>
  </ion-card>
</csaa-csaa-bills-card>
`;
